firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java

```diff
@@ -97,13 +96,15 @@ public class CustomModelDownloadService {
   // TODO(b/258424267): Migrate to go/firebase-android-executors
   @SuppressLint("ThreadPoolCreation")
   public CustomModelDownloadService(
-      FirebaseApp firebaseApp, FirebaseInstallationsApi installationsApi) {
+      FirebaseApp firebaseApp,
+      FirebaseInstallationsApi installationsApi,
+      ExecutorService executorService) {
```

Comment:
Looks like network io is performed on this executor, so bgExecutor is not suitable for this.

Relatedly, down below(lint 164) it looks like a urlconnection is being opened in the caller thread and it should be moved into the continuation of the installationAuthTokenTask. pls move it there.